chore: Remove deprecated NgTabsetModule (#12398)
SAP/spartacus: docs/migration/4_0.md

# Technical Changes in Spartacus 4.0

## Breaking Changes Introduced in 4.0

### StoreFinderComponentsModule

> -  Deprecated import `NgbTabsetModule` has been replaced by `NgbNavModule` for keep compatibility with Bootstrap 8.

Reviewer: ```suggestion
-  Deprecated import `NgbTabsetModule` from `@ng-bootstrap/ng-bootstrap` has been replaced with `NgbNavModule` to support version 8 of the library.
```

### StoreFinderListComponent
- Mobile template has been updated to use nav instead of tabs. [See changes.](https://github.com/SAP/spartacus/pull/12398/files#diff-1db586698a503ea500917fe4d734f84d0729f585aa7c4b56705d9171a38e7f55L64-L120)  

> - Added styling for `ul.nav` for the same appearance.

Reviewer: ```suggestion
- Added styles for `ul.nav` to keep the same appearance.
```
